libmvsim/src/Sensors/LaserScanner.cpp

```diff
@@ -208,9 +206,10 @@ void LaserScanner::simul_post_timestep(const TSimulContext &context)
 
 			callback.m_hit=false;
 			m_world->getBox2DWorld()->RayCast(&callback, sensorPt, endPt);
-			scan.validRange[i] = callback.m_hit ? 1:0;
+			scan.setScanRange(i, maxRange);
+			scan.setScanRangeValidity(i, callback.m_hit);
 
-			float &range = scan.scan[i];
+			float range = scan.scan[i];
```

Comment:
This is the only error in your otherwise great code update!

This was a reference as a way to write the value of the range back into the vector from ::hypotf() or the maxRange a few lines below...

You could change it like as in the next comments: